submodules/PeerInfoUI/Sources/DeviceContactInfoController.swift

```diff
@@ -1072,7 +1072,8 @@ public func deviceContactInfoController(context: AccountContext, updatedPresenta
             if let editingName = state.editingState?.editingName, case let .personName(firstName, lastName, _) = editingName, (!firstName.isEmpty || !lastName.isEmpty) {
                 var urls = filteredData.urls
                 if let createForPeer = createForPeer {
-                    let appProfile = DeviceContactUrlData(appProfile: createForPeer.id)
+                    let addressName: String = createForPeer.addressName ?? ""
+                    let appProfile = DeviceContactUrlData(addressName: addressName)
```

Comment:
It looks like you're using username instead of a raw numeric user ID, so link may invalidate because username may change, while user ID stays the same so I would recommend using it instead

Reply:
or maybe this is supposed to work like this, idk

Reply:
Up: I did some research and it seems like other clients (desktop, web) have no support for deep links right now so maybe it is fine to stick with usernames.

Reply:
Yes, username is working fine currently. So this pr should be good to go then?

Reply:
Well, this code was using deep links before, so maybe there was a reason for it. Anyways now it seems ok to me so let's wait for the others to decide (as I'm not a contributor).